### PR TITLE
Removes Stu3 references from CosmosDb provider

### PR DESCRIPTION
**Pull request by brendankowitz**
## Description
Makes CosmosDb agnostic to FHIR version

## Related issues
Addresses [AB#69080](https://microsofthealth.visualstudio.com/f8da5110-49b1-4e9f-9022-2f58b6124ff9/_workitems/edit/69080).

### CODE REVIEW

File: src/Microsoft.Health.Fhir.Core/Extensions/EnumExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+namespace Microsoft.Health.Fhir.Core.Extensions
+{
+    public static class EnumExtensions
+    {
+        public static T GetValueByEnumLiteral<T>(this string value)
+            where T : Enum
+        {
+            FieldInfo val = typeof(T).GetFields()
+                .FirstOrDefault(x => x.GetCustomAttributes().OfType<EnumLiteralAttribute>().Any(y => y.Literal == value));
```
Comment (brendankowitz) [inline src/Microsoft.Health.Fhir.Core/Extensions/EnumExtensions.cs]: Any should be `Single`

File: src/Microsoft.Health.Fhir.Core/Features/Conformance/IListedCapabilityStatement.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+        bool Experimental { get; set; }
+
+        string Publisher { get; set; }
+
+        string FhirVersion { get; set; }
+
+        IList<string> Format { get; }
+
+        void TryAddRestInteraction(string resourceType, string value);
```
Comment (brendankowitz) [inline src/Microsoft.Health.Fhir.Core/Features/Conformance/IListedCapabilityStatement.cs]: `string value` should be `string interaction`

File: src/Microsoft.Health.Fhir.Stu3.Core/Features/Conformance/ListedCapabilityStatement.cs [hunk 3]
```diff
@@ -42,5 +45,41 @@ public sealed class ListedCapabilityStatement
...
+        {
+            EnsureArg.IsNotNullOrEmpty(value, nameof(value));
+            EnsureArg.IsTrue(Enum.TryParse<ResourceType>(resourceType, out var resource), nameof(resourceType));
+
+            var interaction = value.GetValueByEnumLiteral<TypeRestfulInteraction>();
+
+            this.TryAddRestInteraction(resource, interaction);
+        }
+
+        public void TryAddRestInteraction(string interaction)
```
Comment (brendankowitz) [inline src/Microsoft.Health.Fhir.Stu3.Core/Features/Conformance/ListedCapabilityStatement.cs]: systemInteraction

File: src/Microsoft.Health.Fhir.Api/Modules/FhirModule.cs [hunk 4]
```diff
@@ -159,7 +158,7 @@ ResourceElement SetMetadata(Resource resource, string versionId, DateTimeOffset
                 .AsSelf()
                 .AsService<IProvideCapability>();
 
-            services.TypesInSameAssemblyAs<IProvideCapability>()
+            services.TypesInSameAssembly(typeof(IProvideCapability).Assembly, typeof(Stu3ModelInfoProvider).Assembly)
```
Comment (johnstairs) [inline src/Microsoft.Health.Fhir.Api/Modules/FhirModule.cs]: Temporary, I assume

File: src/Microsoft.Health.Fhir.Stu3.Core/Features/Search/SearchOptionsFactory.cs [hunk 5]
```diff
@@ -120,7 +121,7 @@ public SearchOptions Create(string compartmentType, string compartmentId, string
 
             if (!string.IsNullOrWhiteSpace(resourceType))
             {
-                searchExpressions.Add(Expression.SearchParameter(_resourceTypeSearchParameter, Expression.Equals(FieldName.TokenCode, null, resourceType)));
+                searchExpressions.Add(Expression.SearchParameter(_resourceTypeSearchParameter.ToInfo(), Expression.Equals(FieldName.TokenCode, null, resourceType)));
```
Comment (johnstairs) [inline src/Microsoft.Health.Fhir.Stu3.Core/Features/Search/SearchOptionsFactory.cs]: At some point, we'll want to cache the infos

File: src/Microsoft.Health.Fhir.ValueSets/SystemRestfulInteraction.cs [hunk 6]
```diff
@@ -0,0 +1,18 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+namespace Microsoft.Health.Fhir.ValueSets
+{
+    public static class SystemRestfulInteraction
+    {
+        public const string Transaction = "transaction";
```
Comment (brandonpollett) [inline src/Microsoft.Health.Fhir.ValueSets/SystemRestfulInteraction.cs]: nit: can we sort these?

File: src/Microsoft.Health.Fhir.Api/Modules/FhirModule.cs (hunk 4 above)
Comment (brendankowitz) [reply src/Microsoft.Health.Fhir.Api/Modules/FhirModule.cs]: Hopefully...I should also put these in a property / name them better. i.e. `KnownAssemblies.Core` `KnownAssemblies.CoreStu3`

**Pull request merged**